src/components/organisms/CreditsHistorySection/CreditsHistorySection.tsx

```diff
@@ -0,0 +1,199 @@
+import { FC, useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Center, Container, Grid, GridItem, HStack, Skeleton, Stack, StackDivider, Tag, TagCloseButton, TagLabel, Text, VStack } from '@chakra-ui/react'
+
+import { useActions, useAppState, useEffects } from '@/overmind'
+import { TimeframesData } from '@/@types/Timeframe'
+import { SubRegion } from '@/@types/geojson'
+import { generateCountryByRegion } from '@/utils/GenerateCountryByRegion'
+
+import SubregionIndicator from '@/components/atoms/SubregionIndicator/SubregionIndicator'
+import CreditsHistoryStat from '@/components/atoms/CreditsHistoryStat/CreditHistoryStat'
+import { AutoComplete } from '@/components/molecules/AutoComplete/AutoComplete'
+import CreditsHistoryChart from '@/components/molecules/CreditsHistoryChart/CreditsHistoryChart'
+import { Aeonik } from '@/styles/fonts'
+
+const CreditsHistorySection: FC = () => {
+  const { getCreditsHistory, setCountry, setSubRegion, setTimeframe } = useActions().creditsHistory
+  const { getCreditsHistory: getCreditsHistoryEffect } = useEffects().creditsHistory
+  const { filteredCreditsHistory, dataFilters, rawCreditsHistory } = useAppState().creditsHistory
+  const { t } = useTranslation(`home`)
+  const { t: countryTranslate } = useTranslation(`countries`)
+
+  useEffect(() => {
+    if (!filteredCreditsHistory) {
+      getCreditsHistoryEffect().then((result) => {
+        getCreditsHistory(result)
+      })
+    }
+  }, [])
+
+  const getSearchItems = (
+    subRegion: SubRegion,
+  ): {
+    value: string
+    label: string
+  }[] => {
+    if (subRegion !== SubRegion.WORLD) {
+      return generateCountryByRegion(subRegion).map((country) => ({
+        value: country,
+        label: countryTranslate(`${country}`),
+      }))
+    } else {
+      return Object.values(SubRegion).map((region) => ({
+        value: region,
+        label: t(`regions.${region}`),
+      }))
+    }
+  }
```

Comment:
This could be moved outside the component so that it is not re-recreated on each render.

Reply:
You could use `useMemo` for that?